extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java

```diff
@@ -0,0 +1,225 @@
+package io.quarkus.kogito.deployment;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.drools.compiler.commons.jci.compilers.CompilationResult;
+import org.drools.compiler.commons.jci.compilers.JavaCompiler;
+import org.drools.compiler.commons.jci.compilers.JavaCompilerSettings;
+import org.drools.compiler.compiler.io.memory.MemoryFileSystem;
+import org.drools.modelcompiler.builder.JavaParserCompiler;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.kie.kogito.codegen.ApplicationGenerator;
+import org.kie.kogito.codegen.GeneratedFile;
+import org.kie.kogito.codegen.process.ProcessCodegen;
+import org.kie.kogito.codegen.rules.RuleCodegen;
+
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ArchiveRootBuildItem;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.LiveReloadBuildItem;
+import io.quarkus.runtime.LaunchMode;
+
+public class KogitoAssetsProcessor {
+
+    private final transient String generatedClassesDir = System.getProperty("quarkus.debug.generated-classes-dir");
+
+    @BuildStep(providesCapabilities = "io.quarkus.kogito")
+    FeatureBuildItem featureBuildItem() {
+        return new FeatureBuildItem("kogito");
```

Comment:
Last comment from my side: could you add this as a constant in `FeatureBuildItem` instead?

The list is (mostly) alphabetically ordered.

Reply:
done